src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -107,19 +107,18 @@ static LogicalResult shapeHelperInferMultipleShapes(
 // Handle shape inference for numpy style broadcasting operators.
 template <class OP, class ADAPTOR>
 static LogicalResult inferShapeForBroadcastingOps(
-    OP *op, Type elementType = nullptr) {
-  assert(op && "Expecting a non-null pointer");
-  ADAPTOR operandAdaptor(*op);
+    OP &op, Type elementType = nullptr) {
```

Comment:
Note: passing `op` as a reference allows us to avoid assertion check that op is a valid pointer.